stubs/tensorflow/tensorflow/keras/callbacks.pyi

```diff
@@ -0,0 +1,190 @@
+from _typeshed import Incomplete
+from collections.abc import Callable, Iterable, Mapping, Sequence
+from typing import Any, Literal
+from typing_extensions import TypeAlias
+
+import tensorflow as tf
+from requests.api import _HeadersMapping
+from tensorflow.keras import Model
+from tensorflow.keras.optimizers.schedules import LearningRateSchedule
+
+_Logs: TypeAlias = Mapping[str, Any] | None | Any
+
+class Callback:
+    model: Model  # Model[Any, object]
+    params: dict[str, Any]
+    def set_model(self, model: Model) -> None: ...
+    def set_params(self, params: dict[str, Any]) -> None: ...
+    def on_batch_begin(self, batch: int, logs: _Logs = None) -> None: ...
+    def on_batch_end(self, batch: int, logs: _Logs = None) -> None: ...
+    def on_epoch_begin(self, epoch: int, logs: _Logs = None) -> None: ...
+    def on_epoch_end(self, epoch: int, logs: _Logs = None) -> None: ...
+    def on_predict_batch_begin(self, batch: int, logs: _Logs = None) -> None: ...
+    def on_predict_batch_end(self, batch: int, logs: _Logs = None) -> None: ...
+    def on_predict_begin(self, logs: _Logs = None) -> None: ...
+    def on_predict_end(self, logs: _Logs = None) -> None: ...
+    def on_test_batch_begin(self, batch: int, logs: _Logs = None) -> None: ...
+    def on_test_batch_end(self, batch: int, logs: _Logs = None) -> None: ...
+    def on_test_begin(self, logs: _Logs = None) -> None: ...
+    def on_test_end(self, logs: _Logs = None) -> None: ...
+    def on_train_batch_begin(self, batch: int, logs: _Logs = None) -> None: ...
+    def on_train_batch_end(self, batch: int, logs: _Logs = None) -> None: ...
+    def on_train_begin(self, logs: _Logs = None) -> None: ...
+    def on_train_end(self, logs: _Logs = None) -> None: ...
+
+# A CallbackList has exact same api as a callback,
```

Comment:
But the stub doesn't have the `model` and `params` attribute that exist on `Callback`. Should these be added?

Reply:
They are not documented, but they do exist, so I added them in 75773a5b9.